generators/component/index.js

```diff
@@ -1,19 +1,36 @@
+const R = require('ramda');
+const path = require('path');
 const Base = require('../../lib/Base');
 
-
 module.exports = class extends Base {
-
-  get prompts() {
-    return ['modulesDir', 'moduleName', 'componentName'];
+  get requiredProps() {
+    return ['modulePath', 'componentName'];
   }
 
   prompting() {
     return super.prompting();
   }
 
   writing() {
-    this.copyComponent();
-    this.copyComponent(true);
-  }
+    // To use this in our pipline, we need to return `params`
+    const copyComponentPipe = (params) => {
+      this.copyComponent(params);
+      return params;
+    };
 
+    R.pipe(
+      R.pick(['modulePath', 'componentName']),
+      // Generate presentation component
+      R.merge(R.__, {
```

Comment:
In this case, it's for consistency with the latter case (I agree that this could be an `evolve`). In the latter case, I want to override and update.